packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -22,7 +22,12 @@ export type FormFieldConfig<T> = (
     | { type: "date" }
     // TODO | { type: "dateTime" }
     | { type: "staticSelect"; values?: string[] }
-    | { type: "asyncSelect"; rootQuery: string; labelField?: string }
+    | {
+          type: "asyncSelect";
+          rootQuery: string;
+          labelField?: string;
+          filter?: { name: string; type: "field" | "prop"; gqlVarName: string; gqlVarType: "rootProp" | "filter" };
```

Comment:
that looks good. however gqlVarName and gqlVarType are hard to gasp.

maybe we could infer stuff: gqlVarName can default to name, and gqlVarType can be inferred from Args